pysat/tests/test_meta.py

```diff
@@ -226,1375 +344,1474 @@ def test_init_labels_w_int_default(self):
 
-        # Test transferred
-        assert self.testInst.new_attribute == 'hello'
+        assert self.meta['profiles']['children']['density', 'units'] == 'A'
+        assert self.meta['profiles']['children']['density', 'desc'] == ''
 
-        # Ensure transferred attributes are removed
-        with pytest.raises(AttributeError):
-            self.meta.new_attribute
+        for dvar in ['dummy_str', 'dummy_ustr']:
+            assert self.meta['profiles']['children'][dvar, 'desc'] == 'are fun'
+            assert self.meta['profiles']['children'][dvar, 'bananas'] == 2
         return
 
-    def test_transfer_attributes_to_instrument_leading_(self):
-        """Ensure private custom meta attributes not transferred."""
-
-        self.meta.mutable = True
+    def test_inst_assign_from_meta_w_ho(self):
+        """Test assignment to Instrument from Meta with higher order data."""
+        # I DON'T THINK THIS TEST DOES ANYTHING NEW, WHY IS IT HERE?
 
-        # Set private attributes
-        self.meta._yo_yo = 'yo yo'
-        self.meta.__yo_yo = 'yo yo'
+        self.set_meta(inst_kwargs={'platform': 'pysat', 'name': 'testing'})
+        frame = pds.DataFrame({fkey: np.arange(10) for fkey in self.frame_list},
+                              columns=self.frame_list)
+        self.meta = pysat.Meta()
+        self.meta['dummy_frame1'] = {'units': 'A'}
+        self.meta['dummy_frame2'] = {'desc': 'nothing'}
+        self.dval = 'help'
+        self.testInst[self.dval] = {
+            'data': [frame] * self.testInst.index.shape[0],
+            'units': 'V', 'long_name': 'The Doors', 'meta': self.meta}
+        self.testInst['help2'] = self.testInst[self.dval]
+        self.testInst.meta['help2'] = self.testInst.meta[self.dval]
+
+        assert self.testInst.meta[self.dval].children['dummy_frame1',
+                                                      'units'] == 'A'
+        assert self.testInst.meta['help2', 'long_name'] == 'The Doors'
+        testing.assert_list_contains(self.frame_list,
+                                     self.testInst.meta.ho_data[self.dval])
+        testing.assert_list_contains(self.frame_list,
+                                     self.testInst.meta[self.dval]['children'])
+        for label in ['units', 'desc']:
+            assert self.testInst.meta[self.dval][
+                'children'].hasattr_case_neutral(label)
 
-        # Include standard parameters as well
-        self.meta.new_attribute = 'hello'
-        self.meta.transfer_attributes_to_instrument(self.testInst)
+        assert self.testInst.meta['help2']['children']['dummy_frame1',
+                                                       'desc'] == ''
+        assert self.testInst.meta['help2']['children']['dummy_frame2',
+                                                       'desc'] == 'nothing'
+        assert 'children' not in self.testInst.meta.data.columns
+        return
```

Comment:
Can this test be removed? @rstoneback

Reply:
Potentially.  The higher order meta functions will be removed in 3.2.0 (#789)

Reply:
I also think the parametrized tests I streamlined above this basically do the same thing.  I found a lot of duplicate tests.  But these were complicated enough that I want another pair of eyes on it.